react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = ({ email: '' });
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+
+  }
+
+  handleChange() {
+    this.setState({ email: event.target.value });
+  }
+
+  handleSubmit() {
+    console.log('state:', this.state);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label>
+          Email
+        <input value={this.state.email} onChange={this.handleChange}></input>
+        </label>
+        <button type="submit" value="Submit">Sign Up</button>
```

Comment:
The button doesn't need a `value` attribute, only actual `input`s do. Also while it doesn't hurt anything to have it, the `type` attribute isn't required either. The default type of a `button` is `submit`.